packages/components/src/badge/index.tsx

```diff
@@ -0,0 +1,67 @@
+/**
+ * External dependencies
+ */
+import clsx from 'clsx';
+
+/**
+ * WordPress dependencies
+ */
+import { info, caution, error, published } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import type { BadgeProps } from './types';
+import Icon from '../icon';
+
+function Badge( {
+	className,
+	as: Component = 'div',
+	context = 'neutral',
+	children,
+	...props
+}: BadgeProps ) {
```

Comment:
When passing down `...props` to the underlying element, we want to add in the underlying element props as well. We have a utility to do that (more info [here](https://wordpress.github.io/gutenberg/?path=/docs/components-contributing-guidelines--docs#typescript)):

```suggestion
}: WordPressComponentProps< BadgeProps, 'span', false > ) {
```